reconcile/utils/terrascript/cloudflare_client.py

```diff
@@ -0,0 +1,211 @@
+import os
+import tempfile
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from typing import Iterable, Optional
+
+from terrascript import Terrascript, Terraform, Resource
+from terrascript import provider
+
+from reconcile.utils.external_resource_spec import (
+    ExternalResourceSpec,
+    ExternalResourceSpecInventory,
+)
+from reconcile.utils.terrascript.cloudflare_resources import _CloudflareZoneResource
+
+TMP_DIR_PREFIX = "terrascript-cloudflare-"
+
+
+@dataclass
+class CloudflareAccountConfig:
+    name: str
+    email: str
+    api_key: str
+    account_id: str
+
+
+@dataclass
+class S3BackendConfig:
+    access_key: str
+    secret_key: str
+    bucket: str
+    key: str
+    region: str
+
+
+def create_terrascript_cloudflare(
+    account_config: CloudflareAccountConfig, backend_config: S3BackendConfig
+) -> Terrascript:
+    terrascript = Terrascript()
+
+    terrascript += Terraform(
+        required_providers={
+            # TODO: pull this from account file
+            "cloudflare": {"source": "cloudflare/cloudflare", "version": "3.18"}
+        }
+    )
```

Comment:
Would it make sense to leave it to the user to pass in the desired version?

This is what I am playing with locally where the integration code pulls the provider version from the account file and passes it in to the updated method code below:
```suggestion
def create_terrascript_cloudflare(
    account_config: CloudflareAccountConfig,
    backend_config: S3BackendConfig,
    provider_version: str,
) -> Terrascript:
    terrascript = Terrascript()

    terrascript += Terraform(
        required_providers={
            # TODO: pull this from account file
            "cloudflare": {
                "source": "cloudflare/cloudflare",
                "version": provider_version,
            }
        }
    )
```

Reply:
Yes, that's kind of what I meant to capture with ` # TODO: pull this from account file` -- we'd get this from the account query and inject it in here. There's no reason that I didn't do it already other than that this was just meant to be a placeholder to show that we can offload the creation of `Terrascript` to a separate function rather than including it in the client like we're doing in `TerrascriptClient`.